addons/base_address_city/models/res_partner.py

```diff
@@ -32,10 +32,12 @@ def _fields_view_get_address(self, arch):
         if doc.xpath("//field[@name='city_id']"):
             return arch
 
+        # only add 'parent_id' invisible field if not already available in arch
+        add_parent_id = not doc.xpath("//field[@name='parent_id']")
```

Comment:
maybe ``parent_id_exist = doc.xpath("//field[@name='parent_id']")`` and use it as inverse condition ?